src/diffusers/models/embeddings.py

```diff
@@ -806,6 +920,27 @@ def forward(self, caption):
         return hidden_states
 
 
+# YiYi notes: combine PixArtAlphaTextProjection and HunYuanTextProjection?
```

Comment:
cc @DN6 here should we use `PixArtAlphaTextProjection`instead? it seems like this projection layer is going to be used a lot

Reply:
Yeah I think we can make it a generic TextProjection class and reuse it.